### PR TITLE
docs: One-Sided communication example

### PR DESCRIPTION
**Pull request by lefarov**
Added two examples for One-Sided communication.

### CODE REVIEW

**Comment by lefarov** [comment]
Seems like this test is failing:
```bash
The command "mpiexec -n 1  python $PWD/demo/futures/test_futures.py -v" exited with 15.
```
I hope this problem was not introduce by me adding examples to documentation :)

**Comment by leofang** [comment]
Looks like it only happens with ppcl64le + mpich. Something might be wrong with `getname`. I manually restarted the failed jobs but I don't think it's just random glitches. My guess is it's being fixed in #37.

**Comment by lefarov** [comment]
@leofang should I then wait till it's merged and rebase?

**Comment by dalcinl** [comment]
@lefarov There are a few things I do not like about your code. Give me some time to add questions/suggestions. I'm working on #50 to fix once and for all builds issues on CI worker images.

**Comment by lefarov** [comment]
@dalcinl sure, no rush

**Comment by dalcinl** [comment]
@lefarov I'm quite pedantic about code and documentation quality/accuracy/conciseness. I've modified your examples to make them more in line of what I believe is better code for tutorial material (minimal dependencies, conciseness, naming consistent with other examples within mpi4py, etc.). There was nothing intrinsically wrong with your original code. As I said, I'm just pedantic and opinionated. Ask @leofang in private, he surely has awesome words to say about me :laughing:.

Hopefully, my update still capture your original intentions. If you agree with my changes, please copy over snippets from the gists below, fixup your commit, and rebase to latest master. If you have suggestions for further improvements, let me know.

1. [tut-win-1-py](https://gist.github.com/dalcinl/0c33f1949e70c72daa9f230cc3768ed2)
2. [tut-win-2-py](https://gist.github.com/dalcinl/3ff633f6de6aae25f20ac5276cc36b90)

Note the use of `mpi4py.util.dtlib`, which is a relatively recent addition to master. I removed your `time.sleep()` calls in favor of `Barrier()` to enforce the required ordering. Also note that the second example is passing `disp_unit` to the window constructor such that you can specify offsets in element units (rather than byte units in your original example).

**Comment by lefarov** [comment]
Hi @dalcinl, no worries, being pedantic about the code and documentation makes total sense to me :). Yes, I guess your examples convey the original idea and fit better to the rest of the documentation. I also like the idea of using `Barrier()` instead of sleeping. Though, it seems like my contribution in this PR boils down to encouraging you to write this example :sweat_smile: 

What do you mean by
>fixup my commits

Split one commit into two (one for each example)? Better commit names?

**Comment by dalcinl** [comment]
> it seems like my contribution in this PR boils down to encouraging you to write this example

I put the aesthetics, you put the soul. Ultimately, *L'essentiel est invisible pour les yeux.*

By "fixup your commits" I meant that you should apply modify your examples with my proposed snippets (most likely adding new local commits in your working code, and perhaps using the `--fixup` option), and then re-create a single commit (most likely using `git rebase -i --autosquash` to meld everything in a single commit), and finally `git push --force`. This will update the MR. All this is everyday git hackery for me (I do care about code, docs, and also git history!), but if you are no so much used to do this, please [read about it](https://jordanelver.co.uk/blog/2020/06/04/fixing-commits-with-git-commit-fixup-and-git-rebase-autosquash/), you will not regret learning these git history-editing tricks. PS: Before all that, please run `git fetch && git rebase origin/master` to rebase to latest master.

**Comment by lefarov** [comment]
Thanks, @dalcinl. This git sugar is indeed useful! :)

**Comment by dalcinl** [comment]
Your PR title is much better than the commit message! Could you just change the commit message to "docs: One-Sided communication example" ?

File: docs/source/usrman/tutorial.rst [hunk 1]
```diff
@@ -371,6 +371,73 @@ CUDA-aware MPI + Python GPU arrays
 
    assert cp.allclose(recvbuf, sendbuf*size)
 
+One-Sided Communications
```
Comment (leofang) [inline docs/source/usrman/tutorial.rst:374]: nitpick: need two blank lines to separate sections
```suggestion

One-Sided Communications
```

**Comment by dalcinl** [comment]
@lefarov This is ready to go. Thanks!

PS: @leofang This PR made me remember that we have to add docs generation to our CI.

**Pull request merged**